ui/components/app/assets/asset-list/network-filter/network-filter.tsx

```diff
@@ -0,0 +1,144 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { setTokenNetworkFilter } from '../../../../../store/actions';
+import {
+  getCurrentChainId,
+  getCurrentNetwork,
+  getIsTestnet,
+  getPreferences,
+  getSelectedInternalAccount,
+  getShouldHideZeroBalanceTokens,
+  getNetworkConfigurationsByChainId,
+} from '../../../../../selectors';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { SelectableListItem } from '../sort-control/sort-control';
+import { useAccountTotalFiatBalance } from '../../../../../hooks/useAccountTotalFiatBalance';
+import { Text } from '../../../../component-library/text/text';
+import {
+  Display,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../../../helpers/constants/design-system';
+import { Box } from '../../../../component-library/box/box';
+import { AvatarNetwork } from '../../../../component-library';
+import UserPreferencedCurrencyDisplay from '../../../user-preferenced-currency-display';
+import { CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP } from '../../../../../../shared/constants/network';
+
+type SortControlProps = {
+  handleClose: () => void;
+};
+
+const NetworkFilter = ({ handleClose }: SortControlProps) => {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const chainId = useSelector(getCurrentChainId);
+  const selectedAccount = useSelector(getSelectedInternalAccount);
+  const currentNetwork = useSelector(getCurrentNetwork);
+  const allNetworks = useSelector(getNetworkConfigurationsByChainId);
+  const isTestnet = useSelector(getIsTestnet);
+  const { tokenNetworkFilter, showNativeTokenAsMainBalance } =
+    useSelector(getPreferences);
+  const shouldHideZeroBalanceTokens = useSelector(
+    getShouldHideZeroBalanceTokens,
+  );
+
+  const { totalFiatBalance: selectedAccountBalance } =
+    useAccountTotalFiatBalance(selectedAccount, shouldHideZeroBalanceTokens);
+
+  // TODO: fetch balances across networks
+  // const multiNetworkAccountBalance = useMultichainAccountBalance()
+
+  const handleFilter = (chainFilters: Record<string, boolean>) => {
+    dispatch(setTokenNetworkFilter(chainFilters));
+
+    // TODO Add metrics
+    handleClose();
+  };
+
+  return (
+    <>
+      <SelectableListItem
+        isSelected={!Object.keys(tokenNetworkFilter).length}
+        onClick={() => handleFilter({})}
+      >
+        <Box
+          display={Display.Flex}
+          justifyContent={JustifyContent.spaceBetween}
+        >
+          <Box>
+            <Text
+              variant={TextVariant.bodyMdMedium}
+              color={TextColor.textDefault}
+            >
+              {t('allNetworks')}
+            </Text>
+            <Text
+              variant={TextVariant.bodyMdMedium}
+              color={TextColor.textDefault}
+            >
+              $1,000.00
+            </Text>
+          </Box>
+          <Box display={Display.Flex}>
+            {Object.values(allNetworks).map((network, index) => {
```

Comment:
what about:
```
Object.values(allNetworks).slice(0, 5).map(({ chainId }, index) => {
  const networkImageUrl = CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP[chainId as keyof typeof CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP];

  return (
    <AvatarNetwork
      key={chainId}
      name="All"
      src={networkImageUrl ?? undefined}
      // overlap the icons
      style={{
        marginLeft: index === 0 ? 0 : '-20px',
        zIndex: 5 - index,  // Since we're limiting to 5 networks, zIndex is 5 - index
      }}
    />
  );
});
```

Reply:
Good callout, addressed here: https://github.com/MetaMask/metamask-extension/pull/27884/commits/a2074cd2456cb99db7f3622f316f3bcf220ca70d

Reply:
And here: https://github.com/MetaMask/metamask-extension/pull/27884/commits/836a11b90f5fc4e640d5cfe8dbc99418d90e5c4a